pkg/runctl/run_manager.go

```diff
@@ -568,6 +568,33 @@ func (c *runManager) createTektonTaskRun(ctx *runContext) error {
 	return err
 }
 
+func (c *runManager) addTektonTaskRunParamsForImage(
+	pipelineRun k8s.PipelineRun,
+	tektonTaskRun *tekton.TaskRun,
+) {
+	spec := pipelineRun.GetSpec()
+	jfrSpec := spec.JenkinsfileRunner
+	image := c.pipelineRunsConfig.Image
+	imagePullPolicy := c.pipelineRunsConfig.ImagePullPolicy
+
+	if jfrSpec != nil {
+		if jfrSpec.Image != "" {
+			image = jfrSpec.Image
+			if jfrSpec.ImagePullPolicy == "" {
+				imagePullPolicy = "IfNotPresent"
+			}
+		}
+		if jfrSpec.ImagePullPolicy != "" {
+			imagePullPolicy = jfrSpec.ImagePullPolicy
+		}
```

Comment:
This must be removed, because the pull policy belongs to the respective image and must not be applied to another image:
```suggestion
```

Reply:
The idea is to overwrite the imagePullPolicy of default image wihout specifying the default image again.

Reply:
Example: The Steward admin configures a pull policy `Never`, because the respective image comes from a private registry and is pulled by the image preloader. If the client would set it to `Always`, the image pull will always fail due to a missing image pull secret.

Example: The administrator configures an image tag that gets updated regularly and want new pipeline runs to always use the latest version. Therefore he configures `Always` as pull policy. If a client doesn't know about it and sets the image pull policy to `IfNotPresent`, it may run with outdated images.

Overall I think `image` and `imagePullPolicy` belong together and should not be mixed between different configurations.